### DISCUSSION TITLE
Minimal runtimes for win-x86

### DISCUSSION THREAD

**Discussion by sindizzy** [MaxRev-Dev/gdal.netcore#95] [Q&A] [answered]
Once again thank you for the great repos!!! This has truly made my life a lot easier.

So not sure how ask this as I'm a little vague on how the whole runtimes packages are compiled. But here goes...
In your MaxRev.Gdal.WindowsRuntime.Minimal package it appears all the runtimes get put in the win-x64 folder so I am assuming they got compiled against x64. Now I started my project as AnyCPU but today found I have to use a DLL that only supports x86 so I have to switch my project to the x86 platform. Would you happen to have any advice on how to create a collection of win-x86 drivers?

**Comment by MaxRev-Dev** [comment, marked as answer]
You are right. These packages are compiled and work only on the x64 systems.

If you want to compile for x86 - this brings some complexity into the packaging system, and of course, everything should be tested twice. 
I did not support x86 because it's similar to that issue when we try to stick to some older systems with an old libc.
And in the result, this will be used by several developers only, I assume in total <1%. Wasted efforts.

But if you are enthusiastic and you want to spend some time on this, I can specify some points on compiling for x86 (I hope I did not forget anything):

Shared (vcpkg):
- All triplets must be changed to x86, and you must ensure that every dependency compiles without issues
- Changes to Visual Studio generated files target platform.
- Bootstrapper code must be changed to match the `win-any` and `unix-any` RID folders.

Windows runtime:
- This [vcpkg script](https://github.com/MaxRev-Dev/gdal.netcore/blob/master/win/vcpkg-makefile.vc) should be changed respectively to reflect x86 triplets
- GDAL SDK (main set of dependencies) - https://gisinternals.com/sdk.php
- [Dependency collection script](https://github.com/MaxRev-Dev/gdal.netcore/blob/master/win/collect-deps-makefile.vc) adjustments to find all necessary deps for x86.

Linux runtime:
- This [vcpkg script](https://github.com/MaxRev-Dev/gdal.netcore/blob/master/unix/vcpkg-makefile) should be changed respectively to reflect x86 triplets
- [Dependency collection script](https://github.com/MaxRev-Dev/gdal.netcore/blob/master/unix/collect-deps-makefile) adjustments to find all necessary deps for x86.

**Reply by sindizzy** [reply]
Wow thanks for the in-depth info. Let me give this a shot.
